perfkitbenchmarker/benchmarks/hbase_ycsb_benchmark.py

```diff
@@ -0,0 +1,216 @@
+  hadoop.CheckPrerequisites()
+  ycsb.CheckPrerequisites()
+
+
+def _InstallYCSB(vm):
+  """Install YCSB on 'vm'."""
+  vm.Install('ycsb')
+
+
+def _CreateYCSBTable(vm, table_name=TABLE_NAME, family=COLUMN_FAMILY,
+                     n_splits=TABLE_SPLIT_COUNT):
+  # See: https://issues.apache.org/jira/browse/HBASE-4163
+  template_path = data.ResourcePath(CREATE_TABLE_SCRIPT)
+  remote = os.path.join(hbase.HBASE_DIR,
+                        os.path.basename(os.path.splitext(template_path)[0]))
+  vm.RenderTemplate(template_path, remote,
+                    context={'table_name': table_name,
+                             'family': family,
+                             'n_splits': n_splits,
+                             'use_snappy': FLAGS.hbase_use_snappy})
+  # TODO(connormccoy): on HBase update, add '-n' flag.
+  command = "{0}/hbase shell {1}".format(hbase.HBASE_BIN, remote)
+  vm.RemoteCommand(command, should_log=True)
+
+
+def _GetVMsByRole(vms):
+  hbase_vms = vms[:-FLAGS.ycsb_client_vms]
+  return {'vms': vms,
+          'hbase_vms': hbase_vms,
+          'master': hbase_vms[0],
+          'zk_quorum': hbase_vms[:FLAGS.hbase_zookeeper_nodes],
+          'workers': hbase_vms[1:],
+          'loaders': vms[-FLAGS.ycsb_client_vms:]}
+
+
+def Prepare(benchmark_spec):
+  """Prepare the virtual machines to run hadoop.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  vms = benchmark_spec.vms
+  by_role = _GetVMsByRole(benchmark_spec.vms)
+
+  loaders = by_role['loaders']
+  assert loaders, vms
+
+  # HBase cluster
+  hbase_vms = by_role['hbase_vms']
+  assert hbase_vms, vms
+  master = by_role['master']
+  zk_quorum = by_role['zk_quorum']
+  assert zk_quorum
+  workers = by_role['workers']
+  assert workers
+
+  def InstallHBase(vm):
+    vm.Install('hbase')
+  vms = benchmark_spec.vms
+  vm_util.RunThreaded(InstallHBase, vms)
+  hadoop.ConfigureAndStart(master, workers, start_yarn=False)
+  hbase.ConfigureAndStart(master, workers, zk_quorum)
+
+  _CreateYCSBTable(master)
+
+  vm_util.RunThreaded(_InstallYCSB, loaders)
+
+  # Populate hbase-site.xml on the loaders.
+  master.PullFile(
+      vm_util.GetTempDir(),
+      os.path.join(hbase.HBASE_CONF_DIR, HBASE_SITE))
+  for loader_vm in loaders:
+    loader_vm.PushFile(
+        os.path.join(vm_util.GetTempDir(), HBASE_SITE),
+        os.path.join(ycsb.YCSB_DIR, 'hbase-binding', 'conf', HBASE_SITE))
+
+
+def Run(benchmark_spec):
+  """Spawn YCSB and gather the results.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  vms = benchmark_spec.vms
+  loaders = _GetVMsByRole(vms)['loaders']
+  logging.info('Loaders: %s', loaders)
+
+  executor = ycsb.YCSBExecutor('hbase-10')
+
+  metadata = {'ycsb_client_vms': FLAGS.ycsb_client_vms,
+              'num_vms': FLAGS.num_vms,
```

Comment:
maybe 'hbase_cluster_size": max(FLAGS.num_vms, 2) is more descriptive, unless you were just trying to capture the relevant flags